smtp/smtp.js

```diff
@@ -129,25 +129,33 @@ SMTP.prototype = {
       }
     };
 
+    var callbacks = 0;
     var response = function(err, msg) {
       if (err) {
```

Comment:
since the race condition has to do with both error and response firing together, what if instead you just called 

```
self.sock.removeAllListeners('error')
self.sock.removeAllListeners('response')
```

at the beginning of the `response()` function, would that fix it for you? that might be cleaner as it involves less if/else

i apologize for just noticing this now. i spent some more time trying to understand the problem...